textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,311 @@
+TOKEN : { < DICTIONARY : "DICTIONARY" > }
+TOKEN : { < VIEW : "VIEW" > }
+TOKEN : { < SELECT : "SELECT" > }
+TOKEN : { < EXTRACT : "EXTRACT" > }
+TOKEN : { < FROM : "FROM" > }
+TOKEN : { < AS : "AS" > }
+TOKEN : { < LIMIT : "LIMIT" > }
+TOKEN : { < OFFSET : "OFFSET" > }
+TOKEN : { < USING : "USING" > }
+TOKEN : { < ON : "ON" > }
+TOKEN : { < JOIN : "JOIN" > }
+TOKEN : { < WHERE : "WHERE" > }
+TOKEN : { < OUTPUT : "OUTPUT" > }
+TOKEN : { < INSERT : "INSERT" > }
+TOKEN : { < INTO : "INTO" > }
+TOKEN : { < VALUES : "VALUES" > }
+TOKEN : { < TO : "TO" > }
+TOKEN : { < ID : "ID" > }
+
+TOKEN : { <LPAREN: "(" > }
+TOKEN : { < RPAREN: ")" > }
+TOKEN : { < LBRACE: "{" > }
+TOKEN : { < RBRACE: "}" > }
+TOKEN : { < LBRACKET: "[" > }
+TOKEN : { < RBRACKET: "]" > }
+TOKEN : { < SEMICOLON: ";" > }
+TOKEN : { < COMMA: "," > }
+TOKEN : { < DOT: "." > }
+TOKEN : { < STAR : "*" > }
+
+TOKEN : { < ASSIGN: "="> }
+TOKEN : { < LT: "<" > }
+TOKEN : { < GT: ">" > }
+TOKEN : { < EQ: "==" > }
+TOKEN : { < LE: "<=" > }
+TOKEN : { < GE: ">=" > }
+TOKEN : { < NE: "!=" > }
+
+/*
+ *	Literal Tokens
+ */
+TOKEN : { < NUMBER_LITERAL : ("-")? ((["0"-"9"])+ ("." (["0"-"9"])*)? | "." (["0"-"9"])+) > }
+TOKEN : { < STRING_LITERAL : "\"" ( "\\" ~[] | ~["\""] )* "\"" > }
+TOKEN : { < REGEX_LITERAL : "/" ( "\\" ~[] | ~["/"] )* "/" > }
+TOKEN : { < IDENTIFIER : ["a"-"z"] (["a"-"z","0"-"9"])* > }
+
+/*
+ *	Wrappers for Literal Tokens
+ */
+double numberDouble():
+{
+    String numberString;
+}
+{
+    numberString = numberLiteral() { return Double.parseDouble(numberString); }
+}
+int numberInteger():
+{
+    String numberString;
+}
+{
+    numberString = numberLiteral() { return Integer.parseInt(numberString); }
+}
+String numberLiteral():
+{
+    Token token;
+}
+{
+    token = <NUMBER_LITERAL> { return token.image; }
+}
+String regexLiteral():
+{
+    Token token;
+}
+{
+    token = <REGEX_LITERAL> { return token.image.substring(1, token.image.length()-1).replace("\\/", "/"); }
+}
+String stringLiteral():
+{
+    Token token;
+}
+{
+    token = <STRING_LITERAL> { return token.image.substring(1, token.image.length()-1).replace("\\\"", "\""); }
+}
+String identifier():
+{
+    Token token;
+}
+{
+    token = <IDENTIFIER> { return token.image; }
+}
+
+/*
+ *	List of (some) Literal Tokens separated by comma
+ */
+List<String> identifierList():
+{
+    String identif;
+    List<String> identifiers = new ArrayList<String>();
+}
+{
+    identif=identifier() { identifiers.add(identif); }
+    ( <COMMA> identif=identifier() { identifiers.add(identif); } )*
+    { return identifiers; }
+}
+
+
+/*
+ *	Statements Declaration
+ */
+List<Map<String, Object>> statementsMain(Consumer<Map<String,Object>> statementConsumer) : 
+{
+    List<Map<String, Object>> statements = new ArrayList();
+    Map<String, Object> newStatement;
+}
+{
+    (
+        newStatement = statement() { statements.add(newStatement); }
+        { if(statementConsumer!=null)statementConsumer.accept(newStatement); }
+    )*
```

Comment:
Change it to something more like `if(statementConsumer != null) statementConsumer.accept(newStatement);`